test/functional/apps/discover/_discover.js

```diff
@@ -0,0 +1,134 @@
+          common.debug('clickTimepicker');
+          return discoverPage.clickTimepicker();
+        })
+        .then(function () {
+          common.debug('setAbsoluteRange');
+          return discoverPage.setAbsoluteRange(fromTime, toTime);
+        })
+        .then(function () {
+          common.debug('collapseTimepicker');
+          return discoverPage.collapseTimepicker();
+        });
+      });
+
+
+      bdd.describe('query', function () {
+        var queryName1 = 'Query # 1';
+
+        bdd.it('should show correct time range string', function () {
+          var expectedTimeRangeString =
+          'September 19th 2015, 06:31:44.000 to September 23rd 2015, 18:31:44.000';
+          return discoverPage.getTimespanText()
+          .then(function (actualTimeString) {
+            expect(actualTimeString).to.be(expectedTimeRangeString);
+          })
+          .catch(common.handleError(this));
+        });
+
+        bdd.it('save query should show toast message and display query name', function () {
+          var expectedSavedQueryMessage = 'Discover: Saved Data Source "' + queryName1 + '"';
+          this.timeout = 60000;
+          return discoverPage.saveSearch(queryName1)
+          .then(function () {
+            return discoverPage.getToastMessage(2000);
+          })
+          .then(function (toastMessage) {
+            expect(toastMessage).to.be(expectedSavedQueryMessage);
+          })
+          .then(function () {
+            return discoverPage.waitForToastMessageGone();
+          })
+          .then(function () {
+            return discoverPage.getCurrentQueryName();
+          })
+          .then(function (actualQueryNameString) {
+            expect(actualQueryNameString).to.be(queryName1);
+          })
+          .catch(common.handleError(this));
+        });
+
+        bdd.it('load query should show query name', function () {
+          return discoverPage.loadSavedSearch(queryName1)
+          .then(function () {
+            return common.tryForTime(15000, function () {
+              return discoverPage.getCurrentQueryName()
+              .then(function (actualQueryNameString) {
+                expect(actualQueryNameString).to.be(queryName1);
+              });
+            });
+          })
+          .catch(common.handleError(this));
+        });
+
+        bdd.it('should show the correct bar chart', function () {
+          var expectedBarChartData = [0,0,0,0,1.0968749999999972,7.6781250000000085,
+            37.87875,92.210625,108.590625,71.80875,23.54625,4.753124999999997,
+            2.1206249999999898,7.60499999999999,35.319374999999994,85.044375,110.199375,
+            70.05375000000001,23.180625000000006,4.0218750000000085,1.2431250000000063,
+            6.435000000000002,36.416250000000005,88.408125,108.81,
+            69.395625,22.522499999999994,5.4112499999999955,0.29249999999998977,0,0,0,0,0,0,0,0];
+          this.timeout = 80000;
+          return common.sleep(4000)
```

Comment:
What is this waiting for? It seems like a really long time to wait, and a really brittle method. Given my experience with the settings tests, I'd expect this to cause sporadic problems when running on the CI systems.